LICENSE

```diff
@@ -1,2 +1,30 @@
 # TODO: Write license file.
 #  - Take into account the following file: https://github.com/scikit-learn-contrib/project-template/blob/main/LICENSE
+
+Copyright (c) 2016, Vighnesh Birodkar and scikit-learn-contrib contributors
```

Comment:
I think we need to modify this line in some way since right now it implies this repo was created and is owned by the sklearn-contrib team.
I think the license file should reflect that we are the creators and owners of the project, while acknowledging we used the sklearn template repo and following their copyright.
I'm just not sure what is the best/ propper phrasing for this...

Reply:
yeah, I wasn't sure about our and their names. but clearly if they want their copyright to be mentioned we have to keep their names somewhere as well. but I couldnt find an answer, so i figured we need to start a discussion. the copyright literally says "Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer." 
that should mean we have to copy it - and then potentially add something else?

but i couldnt find a definitive answer. i guess we need to ask emmanuel.

Reply:
I read several discussions on the topic.
The answer I saw most often was to keep the original notice and add our names first. I'll link them a bit later.
Some also recommend to add a per-file copyright notice to indicate which files were modified, which were taken as-is, and which are original, but some people said it is unnecessary since git already tracks these changes.

Reply:
According to these disscussion: 
 - [1](https://www.reddit.com/r/opensource/comments/1abrmxd/guide_or_checklist_on_how_to_fork_bsd3clause/)
 - [2](https://softwareengineering.stackexchange.com/questions/240612/forking-bsd-project-and-changing-license)
 - [3](https://softwareengineering.stackexchange.com/questions/234511/what-is-the-best-practice-for-arranging-third-party-library-licenses-paperwork) 
 - [4](https://softwareengineering.stackexchange.com/questions/125836/do-you-have-to-include-a-license-notice-with-every-source-file)

since we're using the BSD license too it should be safe to just add a notice with our names as well, like so:

```suggestion
Copyright (c) 2024, Konstantin Krivokon and Steffen jw
Copyright (c) 2016, Vighnesh Birodkar and scikit-learn-contrib contributors
```

We also need to add the copyright notice for the [osis project](https://github.com/DataScience-LMU/osis_python_repro/blob/main/LICENSE), since we used some of their template files and their copyright notice also reuqires a copy of their notice, but since they're using the MIT license this might be a bit more complicated. 

Maybe we could split this file into 2? First section for the BSD license (since it covers most of the files) and the 2nd section for the MIT license, and since this lisence covers a minority of the files maybe we could list the files that it covers as well?